dymos/transcriptions/pseudospectral/components/pseudospectral_timeseries_output_comp.py

```diff
@@ -0,0 +1,211 @@
+        # the corresponding input variable.  This is used to ensure that we don't need to connect
+        # the same source to this timeseries multiple times.
+        self._sources = {}
+
+        # Used to track conversion factors for instances when one output that relies on an input
+        # from another variable has potentially different units
+        self._units = {}
+        self._conversion_factors = {}
+
+        # Flag to set if no multiplication by the interpolation matrix is necessary
+        self._no_interp = False
+
+    def setup(self):
+        """
+        Define the independent variables as output variables.
+        """
+        igd = self.options['input_grid_data']
+        ogd = self.options['output_grid_data']
+        output_subset = self.options['output_subset']
+
+        if ogd is None:
+            ogd = igd
+
+        if ogd is igd and output_subset == 'all':
+            self._no_interp = True
+
+        self.input_num_nodes = igd.num_nodes
+        self.output_num_nodes = ogd.subset_num_nodes[output_subset]
+
+        # Build the interpolation matrix which maps from the input grid to the output grid.
+        # Rather than a single phase-wide interpolating polynomial, map each segment.
+        # To do this, find the nodes in the output grid which fall in each segment of the input
+        # grid.  Then build a Lagrange interpolating polynomial for that segment
+        L_blocks = []
+        output_nodes_ptau = ogd.node_ptau[ogd.subset_node_indices[output_subset]].tolist()
+
+        for iseg in range(igd.num_segments):
+            i1, i2 = igd.segment_indices[iseg]
+            iptau_segi = igd.node_ptau[i1:i2]
+            istau_segi = igd.node_stau[i1:i2]
+
+            # The indices of the output grid that fall within this segment of the input grid
+            if ogd is igd:
+                optau_segi = iptau_segi
+            else:
+                ptau_hi = igd.segment_ends[iseg+1]
+                if iseg < igd.num_segments - 1:
+                    idxs_in_iseg = np.where(output_nodes_ptau <= ptau_hi)[0]
+                else:
+                    idxs_in_iseg = np.arange(len(output_nodes_ptau))
+                optau_segi = np.asarray(output_nodes_ptau)[idxs_in_iseg]
+                # Remove the captured nodes so we don't accidentally include them again
+                output_nodes_ptau = output_nodes_ptau[len(idxs_in_iseg):]
+
+            # # Now get the output nodes which fall in iseg in iseg's segment tau space.
+            ostau_segi = 2.0 * (optau_segi - iptau_segi[0]) / (iptau_segi[-1] - iptau_segi[0]) - 1
+
+            # Create the interpolation matrix and add it to the blocks
+            L, _ = lagrange_matrices(istau_segi, ostau_segi)
+            L_blocks.append(L)
+
+        if _USE_SPARSE:
+            self.interpolation_matrix = sp.block_diag(L_blocks, format='csr')
+        else:
+            self.interpolation_matrix = block_diag(*L_blocks)
```

Comment:
Since `_USE_SPARSE` is always True, are the conditionals just in here for future testing/benchmarking?

Reply:
Yeah.  Basically to give us a quick way to enable/disable it should the need arise.